src/components/Text/native/Text.js

```diff
@@ -1,29 +1,46 @@
 import PropTypes from 'prop-types';
 import styled from 'styled-components';
+import { Platform } from 'react-native';
+import { theme } from '../../../theme';
+
+const isPlatformAndroid = !(Platform.OS === 'ios');
+
+export const fontFamilies = {
+  regular: isPlatformAndroid ? 'Roboto' : 'SFProDisplay-Regular',
+  medium: isPlatformAndroid ? 'Roboto-Medium' : 'SFProDisplay-Medium',
+  semibold: isPlatformAndroid ? 'Roboto-Medium' : 'SFProDisplay-Semibold',
+  bold: 'Averta-Bold',
+};
+
+export const fontWeights = theme.fontWeight;
 
 const Text = styled.Text`
-  ${(p) => p.color ? `color: ${p.theme.color[p.color]};` : ''}
-  ${(p) => p.size ? `font-size: ${p.theme.fontSize[p.size]};` : ''}
-  ${(p) => p.weight ? `font-weight: ${p.theme.fontWeight[p.weight]};` : ''}
-  ${(p) => {
-    if (p.family) {
-      return `font-family: ${p.family}`;
+  ${(props) => props.color ? `color: ${props.theme.color[props.color]};` : ''}
+  ${(props) => props.size ? `font-size: ${props.theme.fontSize[props.size]};` : ''}
+  ${(props) => props.weight ? `font-weight: ${props.theme.fontWeight[props.weight]};` : ''}
+  ${(props) => props.lineHeight ? `line-height: ${props.lineHeight};` : ''}
+  ${(props) => {
+    if (props.family) {
+      return `font-family: ${fontFamilies[props.family]}`;
     }
-    return `font-family: ${p.theme.fontFamily[p.weight]}`;
+    return `font-family: ${props.theme.fontFamily[props.weight]}`;
   }}
   `;
 
 Text.propTypes = {
   color: PropTypes.string,
   size: PropTypes.string,
   weight: PropTypes.string,
+  lineHeight: PropTypes.number,
```

Comment:
Don't allow lineHeight customization, our style guide rules are:
if it is a single line of text, lineHeight should be default
if it is a paragaph of text, lineHeight should be set to 1.5